active/gridsyntax/GridSyntaxSpec2.md

```diff
@@ -0,0 +1,368 @@
+
+About this "content property", it's not literally a property named "content" (although it can be and [is sometimes](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ContentControl.Content)). This is essentially the default property of a class, and is marked with the [ContentProperty] attribute.
+
+For example the [SolidColorBrush](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) is defined roughly like this (in C# terms):
+
+```cs
+[ContentProperty(Name = "Color")]
+public sealed class SolidColorBrush : Brush
+{
+    ...
+    public Color Color { get; set; }
+}
+```
+
+Because [SolidColorBrush.Color](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush.Color) is marked as the [ContentProperty] (and because a Color can be created from a string), you've always been able to write this short-hand Xaml:
+
+```xml
+<SolidColorBrush>Red</SolidColorBrush>
+```
+
+If it weren't for the [ContentProperty] on SolidColorBrush, you'd have to use the equivalent full syntax (and that's still valid to use if you prefer):
+
+```xml
+<SolidColorBrush>
+    <SolidColorBrush.Color>
+        Red
+    </SolidColorBrush.Color>
+</SolidColorBrush>
+```
+
+## Making things work for Grid
+
+Back to Grid, with the new syntax feature and with ColumnDefinition.Width now marked as the [ContentProperty]:
+
+```cs
+[ContentProperty(Name = "Width")]
+public sealed class ColumnDefinition : DependencyObject
+{
+        public GridLength Width { get; set; }
+}
+```
+
+you can now write
+
+```xml
+<ColumnDefinition>Auto</ColumnDefinition>
+```
+
+(There's already a converter to create a GridLength from a string.)
+
+Putting that all together, back to the original goal:
+
+<Grid ColumnDefinitions="Auto, *">
+
+Two ColumnDefinition objects are created and added to the ColumnDefinitions collection property, which is ready-only but defaults to an empty collection. The first item has its Width set to GridLength.Auto, the second has it's Width set to 1 GridUnitType.Star.
+
+
+## Special characters and escaping
+
+The XML attribute value is interpreted as a comma-separated list of items, which creates a problem if you want to use a comma an item value. You can solve this using quotes, for example, this initializes a collection with two items:
+
+```xml
+<WordWheel Words = " 'Hello, world', 'how is the weather?' " />
+```
+
+(Xaml markup extensions have the same command issue and the same quote syntax. In general the syntax rules follow those of markup extensions.)
+
+Alternatively you can use a backslash as an escape character before the comma:
+
+```xml
+Words = " I weigh 90\,7 KG    "
+```
+| todo: handle apostrophe in "How's the weather?" case
```

Comment:
I remember that we can't use &apos; in XAML. Just to refresh my mind, I'm assuming we can't just escape the apostrophe with \' either? 

In that case, do we just suggest using single quotes on the outside and double quotes for each of your values?

Words = ' "How's the weather", "It's sunny" '

Reply:
There's a bug in markup extension syntax today that prevents the quote from being escaped. But good point, there's no reason to reproduce that bug in this syntax.

Reply:
https://github.com/microsoft/microsoft-ui-xaml/issues/2904